src/api/server.rkt

```diff
@@ -24,24 +39,38 @@
 (define (make-path id)
   (format "~a.~a" id *herbie-commit*))
 
-; Helers to isolated *completed-jobs*
-(define (completed-job? id)
-  (hash-has-key? *completed-jobs* id))
+(define (completed-job? job-id)
+  (define-values (a b) (place-channel))
+  (place-channel-put receptionist (list 'check job-id b))
+  (when verbose
+    (eprintf "Checking if job ~a is completed.\n" job-id))
+  (place-channel-get a))
 
 ; Returns #f is now job exsist for the given job-id
-(define (get-results-for id)
-  (hash-ref *completed-jobs* id #f))
+(define (get-results-for job-id)
+  (define-values (a b) (place-channel))
+  (place-channel-put receptionist (list 'check job-id b))
+  (when verbose
+    (eprintf "Getting result for job: ~a.\n" job-id))
+  (place-channel-get a))
 
-; I don't like how specific this function is but it keeps the API boundary.
-(define (get-improve-job-data)
-  (for/list ([(k v) (in-hash *completed-jobs*)] #:when (equal? (job-result-command v) 'improve))
-    (get-table-data v (make-path k))))
+(define (get-improve-table-data)
+  (define-values (a b) (place-channel))
+  (place-channel-put receptionist (list 'improve b))
```

Comment:
Now that I think about it, this doesn't make sense. The receptionist has to hold the data because it's the only one that's actually active when the job is finished.

Reply:
Ya, we can play around with different ways of storing and retrieving the data, I have a few ideas. I think cleaning up/ deleting a lot of code around how `run-herbie` is called might help make a decision/priorities here more clear.